lib/plugins/aws/package/compile/events/kafka.js

```diff
@@ -0,0 +1,212 @@
+'use strict';
+class AwsCompileKafkaEvents {
+  constructor(serverless) {
+    this.serverless = serverless;
+    this.provider = this.serverless.getProvider('aws');
+
+    this.hooks = {
+      'package:compileEvents': this.compileKafkaEvents.bind(this),
+    };
+
+    const secretsManagerArnRegex =
+      'arn:[a-z-]+:secretsmanager:[a-z0-9-]+:\\d+:secret:[A-Za-z0-9/_+=.@-]+';
+
+    this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'kafka', {
+      type: 'object',
+      properties: {
+        accessConfigurations: {
+          type: 'object',
+          minProperties: 1,
+          properties: {
+            vpcSubnet: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+                pattern: 'subnet-[a-z0-9]+',
+              },
+            },
+            vpcSecurityGroup: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+                pattern: 'sg-[a-z0-9]+',
+              },
+            },
+            saslScram256Auth: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+                pattern: secretsManagerArnRegex,
+              },
+            },
+            saslScram512Auth: {
+              type: 'array',
+              minItems: 1,
+              items: {
+                type: 'string',
+                pattern: secretsManagerArnRegex,
+              },
+            },
+            additionalProperties: false,
```

Comment:
This should be set in same level as `properties` (currently seems as an extra property definition)